app/models/user.rb

```diff
@@ -199,6 +201,28 @@ def username_required?
   def email_required?
     !erased?
   end
+  
+  def has_official_email?
+    domain = Setting.value_for 'email_domain_for_officials'
+    return false if !email or !domain or domain.length == 0
```

Comment:
This line seems to belong to the rake task rather than a check that should be done in every save.

Rake tasks are great to manipulate existing data, but should not add logic to the models that is not commonly used. We usually create rake tasks to manipulate the code and then delete the rake task, as it can be confusing to keep one-time rake tasks around.